caffe2/core/context_gpu.cu

```diff
@@ -178,8 +178,8 @@ static std::unordered_map<void*, uint8_t> g_cuda_device_affiliation;
 // Data structures for optional memory tracking. Access to these structures
 // is guarded by the CUDAContext::mutex.
 static std::unordered_map<void*, long> g_size_map;
-static std::vector<long> g_total_by_gpu_map(C10_COMPILE_TIME_MAX_GPUS, 0);
-static std::vector<long> g_max_by_gpu_map(C10_COMPILE_TIME_MAX_GPUS, 0);
+static std::vector<long> g_total_by_gpu_map(c10::Device::MAX_NUM_DEVICES, 0);
```

Comment:
In line 176, shouldn't `static std::unordered_map<void*, uint8_t> g_cuda_device_affiliation;` also be updated to uint16_t?

Reply:
Good catch, I guess we should use `DeviceIndex` here too.